[quant][pt2e][refactor] Remove `backend_config` from `_maybe_insert_input_observers_for_node`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98094

Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][pt2e][refactor] Remove `backend_config` of `_maybe_insert_input_observers_for_node`

Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][pt2e][refactor] Remove `backend_config` of `_maybe_insert_input_observers_for_node`"


Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
> The goal is to remove the need to use backend_config when pt2e flow code call this function

can you share some context on why?

jerryzh168 commented:
> > The goal is to remove the need to use backend_config when pt2e flow code call this function
> 
> can you share some context on why?

sure, we don't need this in prepare since validation should be done in quantizer.validate(...) function, so even we have this object (or something similar), it should live in quantizer instead

jcaip commented on torch/ao/quantization/fx/utils.py:
```diff
@@ -86,25 +86,23 @@ class ObservedGraphModuleAttrs:
+        bias_index = node.meta["target_dtype_info"].get("bias_index", None)
+    if bias_index is not None and bias_index < len(node.args) and node.args[bias_index] is arg:
+        return True
+    return node.kwargs.get("bias") is arg
```
same here

jcaip reviewed (approved):
LGTM

jerryzh168 commented on torch/ao/quantization/fx/utils.py:
```diff
@@ -86,25 +86,23 @@ class ObservedGraphModuleAttrs:
+        weight_index = node.meta["target_dtype_info"].get("weight_index", None)
+    if weight_index is not None and weight_index < len(node.args) and node.args[weight_index] is arg:
+        return True
+    return node.kwargs.get("weight") is arg
```
this might be needed for fx graph mode quantization

jerryzh168 committed:
Update on "[quant][pt2e][refactor] Remove `backend_config` from `_maybe_insert_input_observers_for_node`"


Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][pt2e][refactor] Remove `backend_config` from `_maybe_insert_input_observers_for_node`"


Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge

jerryzh168 commented:
@pytorchbot merge
